### PR TITLE
ORC-1001: Bump maven-enforcer-plugin to 3.0.0

### PR DESCRIPTION
**Pull request by autumnust**
### What changes were proposed in this pull request?

While working on another PR, I found the head of main failed in GA with the failure like below. 

```Error:  Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce (enforce-java-version) on project orc: Execution enforce-java-version of goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce: java.lang.ExceptionInInitializerError: null
Error:  -----------------------------------------------------
Error:  realm =    plugin>org.apache.maven.plugins:maven-enforcer-plugin:1.4.1
Error:  strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
Error:  urls[0] = file:/Users/runner/.m2/repository/org/apache/maven/plugins/maven-enforcer-plugin/1.4.1/maven-enforcer-plugin-1.4.1.jar
...
Error:  Number of foreign imports: 1
Error:  import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
Error:  
Error:  -----------------------------------------------------: begin 0, end 3, length 2
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
make[2]: *** [java/shims/orc-shims-1.8.0-SNAPSHOT.jar] Error 1
make[1]: *** [java/CMakeFiles/java_build.dir/all] Error 2
make: *** [all] Error 2
Error: Process completed with exit code 2.
```

Google a bit and found a similar issue: https://github.com/apache/accumulo-proxy/pull/19 
This PR is a draft to see if the same change could work here ( since I don't have specific env like failed CI)

### Why are the changes needed?

Recover `compile` and `package` without tests.

### How was this patch tested?

Manual.
```
mvn compile
mvn package -DskipTests
```

### CODE REVIEW

**Comment by guiyanakuang** [comment]
I think the JDK17 General-Availability Release was released two days ago. Github workflow updated JDK17. Now System.getProperty("java.version") returns "17" which causes a lot of program exceptions.

```
Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)
Maven home: /usr/share/apache-maven-3.8.2
Java version: 17, vendor: Azul Systems, Inc., runtime: /opt/hostedtoolcache/jdk/17.0.0/x64
```

For example org.apache.hadoop.util.Shell.java
```
private static boolean IS_JAVA7_OR_ABOVE =
      System.getProperty("java.version").substring(0, 3).compareTo("1.7") >= 0;
```
java.lang.StringIndexOutOfBoundsException: String index out of range: 3

**Comment by dongjoon-hyun** [comment]
Oh, I didn't saw your PR and make a PR to upgrade Java 17.
- https://github.com/apache/orc/pull/911

**Comment by dongjoon-hyun** [comment]
Actually, this is very unfortunate situation because we release Apache ORC 1.7.0. :(

**Review by dongjoon-hyun — COMMENTED** [review]
At least, I verified that this PR recover `compile` and `package` (without tests).
Thank you, @autumnust .

File: java/pom.xml [hunk 1]
```diff
@@ -90,6 +90,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.0.0-M3</version>
```
Comment (dongjoon-hyun) [inline java/pom.xml:93]: We can use `3.0.0`.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @autumnust and @guiyanakuang .
We can fix the issue step by step.

**Comment by guiyanakuang** [comment]
> Actually, this is very unfortunate situation because we released Apache ORC 1.7.0. :(

@dongjoon-hyun, probably JDK official will fix this problem, because there are many programs involved. It should not be necessary for each program to upgrade the Java version judgment. So don't worry about the ORC 1.7.0 compatibility issue.

**Comment by dongjoon-hyun** [comment]
BTW, `maven-enforcer-plugin` issue exists only at `main`. I landed at `main` only.

**Comment by dongjoon-hyun** [comment]
Do you have any reference for that? Actually, Java community is a little notorious because they don't care about that many programs, @guiyanakuang .

**Comment by guiyanakuang** [comment]
@dongjoon-hyun I remember a similar thing happening with JDK 9. I always thought it was the JDK that officially solved the version number problem. I just googled it and I was wrong. : (

**Comment by dongjoon-hyun** [comment]
Got it. Then, that's a big relief to me. :)

**Comment by dongjoon-hyun** [comment]
Oh, I misread the message and delete my previous comment~ If you received them via the email, please forget about that.

**Comment by dongjoon-hyun** [comment]
BTW, this might be only our test framework issue. Let me dig more.

**Comment by guiyanakuang** [comment]
Yes, there is no code directly related to java version in ORC, the example I gave is from hadoop 2.2. It has been fixed at https://issues.apache.org/jira/browse/HADOOP-14586. Don't worry too much.

**Comment by dongjoon-hyun** [comment]
Ya, that could be one reason to raise the `min.hadoop.version`.

BTW, for Java 17 testing, I'll add a `profile` as a separate PR.

**Pull request merged**